2DPlatformerController/Assets/ScoreScript.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+public class ScoreScript : MonoBehaviour
+{
+    public static int scoreValue;
+    Text score;
+    // Start is called before the first frame update
+    void Start()
+    {
+        score = GetComponent<Text>();
```

Comment:
This is fine, but there is another way to do it. GetComponent is pretty harmless if there are only a couple of them and they are called once. It becomes problematic if there are a lot of them and they are in Update or something. In times like those, if the variable is a public field, you can actually click and drag a reference from the hierarchy into the slot in the inspector. Also, if you want to be able to do that but you want the field to be private, you can do
`[SerializeField]
private Text text = null;`
and you'll be able to drag a reference in there as well.
For this particular instance, however, what you have done here is fine.